controlplane/kubeadm/internal/controllers/helpers.go

```diff
@@ -74,12 +74,8 @@ func (r *KubeadmControlPlaneReconciler) reconcileKubeconfig(ctx context.Context,
 		return ctrl.Result{}, errors.Wrap(err, "failed to retrieve kubeconfig Secret")
 	}
 
-	// check if the kubeconfig secret was created by v1alpha2 controllers, and thus it has the Cluster as the owner instead of KCP;
-	// if yes, adopt it.
-	if util.IsOwnedByObject(configSecret, cluster) && !util.IsControlledBy(configSecret, kcp) {
-		if err := r.adoptKubeconfigSecret(ctx, cluster, configSecret, controllerOwnerRef); err != nil {
-			return ctrl.Result{}, err
-		}
+	if err := r.adoptKubeconfigSecret(ctx, cluster, configSecret, kcp); err != nil {
```

Comment:
GIven comments elsewhere I wonder if we want to use "adopt" in the func name

Reply:
No objection from my side. Just saw this comment on a few other PRs, but maybe this case is different

Reply:
I think the difference (from my perspective) is new functions not related to "adoption" vs old functions that were previously named like this.

Reply:
I think I wouldn't care too much about if that func previously was named like this or not.
But if this is an adoption while the others are not it makes sense.

Independent of that, no strong opinions from my side.